packages/Laravel/src/Rector/Class_/InlineValidationRulesToArrayDefinitionRector.php

```diff
@@ -175,10 +128,73 @@ private function transformConcatExpressionToSingleString(Concat $concat): string
 
                 $output .= implode('\\', $name->parts);
             } else {
-                throw new ShouldNotHappenException('Unexpected expression type ' . get_class($expressionPart));
+                // unable to process
+                return null;
             }
         }
 
         return $output;
     }
+
+    /**
+     * @return Expr[]
+     */
+    private function createNewRules(ArrayItem $arrayItem): array
+    {
+        $newRules = $this->transformRulesSetToExpressionsArray($arrayItem->value);
+
+        foreach ($newRules as $key => $newRule) {
+            if (! $newRule instanceof Concat) {
+                continue;
+            }
+
+            $fullString = $this->transformConcatExpressionToSingleString($newRule);
+            if ($fullString === null) {
+                return [];
+            }
+
+            $matchesExist = preg_match('#^exists:(\w+),(\w+)$#', $fullString, $matches);
+            if ($matchesExist === false || $matchesExist === 0) {
+                continue;
+            }
+
+            $ruleClass = $matches[1];
+            $ruleAttribute = $matches[2];
+
+            $newRules[$key] = new StaticCall(
+                new FullyQualified('Illuminate\Validation\Rule'),
+                'exists',
+                [new Arg(new ClassConstFetch(new Name($ruleClass), 'class')), new Arg(new String_($ruleAttribute))]
+            );
+        }
+
+        return $newRules;
+    }
+
+    private function shouldSkipArrayItem(ArrayItem $arrayItem): bool
+    {
+        $classNode = $arrayItem->getAttribute(AttributeKey::CLASS_NODE);
+        if ($classNode === null) {
+            return true;
+        }
+
+        if (! $this->isObjectType($classNode, self::FORM_REQUEST_CLASS)) {
+            return true;
+        }
+
+        $methodNode = $arrayItem->getAttribute(AttributeKey::METHOD_NODE);
+        if ($methodNode === null) {
+            return true;
+        }
+
+        if (! $this->isName($methodNode, 'rules')) {
+            return true;
+        }
+
+        if (! $arrayItem->value instanceof String_ && ! $arrayItem->value instanceof Concat) {
+            return true;
+        }
+
+        return false;
+    }
```

Comment:
@sashabeton I did a small code cleanup of your rule + making coding standard pass, in case you'd be intereseted.

Main take aways:

- use the most precise node possible - `ArrayItem`
- add `shouldSkipX()` (here `shouldSkipArrayItem()`) method to skip invalid nodes in standalone method
- avoid `&$reference` if possible, they create very hard to solve bugs